### PR TITLE
Deploy all

### PR DESCRIPTION
**Pull request by odra**
Adds the required changes to run a single playbook to deploy all services.

Verification:

```sh
ansible-playbook -e "eval_app_host=your-cluster-host"  playbooks/all.yml
``` 

It can also be used with tags:

```
#specify which tags to run
ansible-playbook -e "eval_app_host=your-cluster-host" playbooks/all.yml --tags 'bootstrap,rhsso,che'
#skip some tags
ansible-playbook -e "eval_app_host=your-cluster-host"  playbooks/all.yml --skip-tags 'rhsso'
```

Specify the action (install or uninstall) - only supported in che's role for now:

```sh
ansible-playbook -e "eval_app_host=your-cluster-host" -e 'eval_action=uninstall' -e 'che_delete_namespace=true' playbooks/all.yml --tags 'bootstrap,che'
```

### CODE REVIEW

**Comment by aidenkeating** [comment]
👀

**Comment by aidenkeating** [comment]
@odra Is there any documentation about the variables and which ones I need to set? I'm not sure what `eval_app_host` does for example.

**Comment by odra** [comment]
`eval_app_host` is the cluster host, currently only used by che's role (routing_suffix template var) I will update the readme file to add some explanation about common.yml playbook vars.

File: evals/inventories/group_vars/all/launcher.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+launcher_namespace: "{{ eval_namespace | default('launcher') }}"
```
Comment (aidenkeating) [inline evals/inventories/group_vars/all/launcher.yml]: Apologies, this is a bug from my Launcher work, it shouldn't go in `eval_namespace`. But a user should be able to pick the namespace if they want. Mind changing this to:

```
launcher_namespace: "{{ launcher_namespace | default('launcher') }}"
```

**Comment by pmccarthy** [comment]
@odra When you say cluster host, what are you referring to? The cluster master URL? I wonder if it would be better to have the env var named openshift_master_url or something along those lines as it's more clear what the variable represents. Just a thought.

**Comment by odra** [comment]
It should be the "app/userspace subdomain" hostname which may not always be the same as the master host (such as *.apps.cloudservices.skunkhenry.com).

For example: when using minishift the url should be `$vm_ip.nip.io` ($vm_ip being the master url/host)

**Comment by pmccarthy** [comment]
@odra Ah ok, I wonder in that case if we would be better off with a var named openshift_router_subdomain maybe? Just to avoid any confusion

**Comment by pmccarthy** [comment]
@odra There's an entry in the master-config.yml:
```
routingConfig:
  subdomain: apps.integreatly.openshiftworkshop.com
```
I wonder could we retrieve it from the playbooks? We already need access to this file for rhsso deployments anyway

File: evals/playbooks/all.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
+---
```
Comment (pmccarthy) [inline evals/playbooks/all.yml]: @odra It might be better to rename this playbook to install.yml to be more clear around what the playbook will do

**Comment by pmccarthy** [comment]
@odra I've run the install playbooks:
```
ansible-playbook -i inventories/hosts playbooks/all.yml -e eval_app_host=apps.integreatly.openshiftworkshop.com -e launcher_github_client_id=<id> -e launcher_github_client_secret=<secret> -e che_github_client_id=<id> -e che_github_client_secret=<secret>
```
The components have been installed successfully but for some reason it hasn't updated the rhsso identity provider in the master configs. Appears to have skipped this task: https://github.com/integr8ly/installation/blob/master/evals/roles/rhsso/tasks/identityprovider.yml

**Comment by odra** [comment]
@pmccarthy I need to update my branch, it was a bit outdated. I will also add a task to retrieve the subdmain hostname from the master's config file.

**Comment by pmccarthy** [comment]
@odra You'll need to update the rhsso group vars to reflect changes made in https://github.com/integr8ly/installation/pull/16/files#diff-369b507f19deddd47a5b2df66f074485

At the moment the rhsso install fails due to invalid user credentials:
```
TASK [rhsso : Generate RH-SSO auth token for admin user] *************************************************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"attempts": 1, "changed": false, "connection": "close", "content": "{\"error\":\"invalid_grant\",\"error_description\":\"Invalid user credentials\"}", "content_length": "72", "content_type": "application/json", "date": "Tue, 14 Aug 2018 13:03:35 GMT", "failed": true, "json": {"error": "invalid_grant", "error_description": "Invalid user credentials"}, "msg": "Status code was not [200]: HTTP Error 401: Unauthorized", "redirected": false, "set_cookie": "60c5c0687cc93ff2a7a3ae7197abd3d0=659da5c6815e4789fedacfdf0416c19d; path=/; HttpOnly; Secure", "status": 401, "url": "https://secure-sso-rhsso.apps.integreatly.openshiftworkshop.com/auth/realms/openshift/protocol/openid-connect/token"}
```

**Comment by pmccarthy** [comment]
@odra I temporarily added the correct rhsso vars locally to continue testing but it looks like the vars in the inventory hosts file are not being read in now. Also having issues targetting the master host specified in the inventory file too

```
TASK [rhsso : Add RH-SSO identity provider to master config] *********************************************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"failed": true, "msg": "The task includes an option with an undefined variable. The error was: 'openshift_master_config' is undefined\n\nThe error appears to have been in '/tmp/installation/evals/roles/rhsso/tasks/identityprovider.yml': line 16, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Add RH-SSO identity provider to master config\n  ^ here\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: 'openshift_master_config' is undefined"
```

File: evals/roles/rhsso/defaults/main.yml [hunk 3]
```diff
@@ -10,6 +10,8 @@ rhsso_image_stream_tag: "1.1"
 rhsso_force_image_streams_update: true
 
 rhsso_admin_username: admin
+rhsso_service_username: evals@example.com
```
Comment (pmccarthy) [inline evals/roles/rhsso/defaults/main.yml]: These are no longer required. We basically need to end up with this set of vars: https://github.com/integr8ly/installation/blob/master/evals/roles/rhsso/vars/main.yml

Comment (pmccarthy) [reply evals/roles/rhsso/defaults/main.yml]: If we're moving all of the rhsso env vars to the group_vars dir then we can just leave this file empty

File: evals/inventories/group_vars/all/rhsso.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+rhsso_template_filename: sso72-x509-postgresql-persistent.json
+rhsso_template_url: https://raw.githubusercontent.com/jboss-openshift/application-templates/ose-v1.4.11/sso/
+
+rhsso_image_stream: redhat-sso72-openshift
+rhsso_image_stream_filename: sso72-image-stream.json
+rhsso_image_stream_tag: "1.1"
+rhsso_force_image_streams_update: true
+
+rhsso_admin_username: admin
+rhsso_service_username: evals@example.com
```
Comment (pmccarthy) [inline evals/inventories/group_vars/all/rhsso.yml]: @odra These vars don't seem to match up with master: https://github.com/integr8ly/installation/blob/master/evals/roles/rhsso/vars/main.yml

For example, rhsso_service_username and rhsso_service_password are not required any longer. I think you can just copy/paste what's on master into this vars file

**Comment by pmccarthy** [comment]
@odra For some reason the inventory file is not being read correctly. For example, the ```openshift_master_config``` var set in the hosts file is undefined according to the playbooks

```
TASK [rhsso : Add RH-SSO identity provider to master config] *********************************************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"failed": true, "msg": "The task includes an option with an undefined variable. The error was: 'openshift_master_config' is undefined\n\nThe error appears to have been in '/tmp/installation/evals/roles/rhsso/tasks/identityprovider.yml': line 16, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Add RH-SSO identity provider to master config\n  ^ here\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: 'openshift_master_config' is undefined"}
```

**Comment by aidenkeating** [comment]
:eyes:

**Comment by aidenkeating** [comment]
@odra I'm seeing

```
fatal: [127.0.0.1]: FAILED! => {"failed": true, "msg": "The task includes an option with an undefined variable. The error was: 'openshift_master_config' is undefined\n\nThe error appears to have been in '/home/ec2-user/installation/evals/playbooks/install-all.yml': line 40, column 11, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n        -\n          name: Retrieve app submain from master node\n          ^ here\n\nexception type: <class 'ansible.errors.AnsibleUndefinedVariable'>\nexception: 'openshift_master_config' is undefined"}
```

Should I be setting this myself?

**Comment by pmccarthy** [comment]
@odra I've re-run the install, seeing this error now on launcher install:
```
TASK [launcher : Create launcher namespace] **********************************************************************************************************************************************************************************************************************************************************************************
fatal: [127.0.0.1]: FAILED! => {"failed": true, "msg": "The conditional check 'launcher_check_namespace_exists_cmd.rc != 0' failed. The error was: error while evaluating conditional (launcher_check_namespace_exists_cmd.rc != 0): 'dict object' has no attribute 'rc'\n\nThe error appears to have been in '/tmp/installation/evals/roles/launcher/tasks/create-namespace.yml': line 7, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Create launcher namespace\n  ^ here\n"}
```

**Comment by odra** [comment]
@aidenkeating you need to set both "launcher_github_client_id" and "launcher_github_client_secret" from the cli since those are not set (I removed all pre-defined tokens/hostnames from the roles)

**Comment by pmccarthy** [comment]
@aidenkeating @odra Latest changes have been pushed. Perhaps one of you could test this out on another environment for validation?

Command run:
```
ansible-playbook -i inventories/hosts playbooks/install-all.yml -e launcher_github_client_id=<client-id> -e launcher_github_client_secret=<client-secret> -e che_github_client_id=<client-id> -e che_github_client_secret=<client-secret>
```

**Comment by odra** [comment]
@pmccarthy worked as expected

File: evals/playbooks/install-all.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      -
+        name: Retrieve app subdomain from master node
+        slurp:
+          src: "{{ eval_openshift_master_config_path }}"
+        register: eval_master_config
+        become: yes
+      -
+        add_host:
+          name: EVAL_VARS
+          eval_app_host: "{{ (eval_master_config['content'] | b64decode | from_yaml)['routingConfig']['subdomain'] }}"
```
[the remark this answers is not in the document]
Comment (pmccarthy) [reply evals/playbooks/install-all.yml]: @odra Sure, makes sense. Any articles I found around sharing vars between plays are suggesting this approach

**Pull request merged**